Simbody/src/RigidBodyNode_LoneParticle.cpp

```diff
@@ -148,6 +150,19 @@ void realizeInstance(const SBStateDigest& sbs) const override {
     SBTreeAccelerationCache& ac = sbs.updTreeAccelerationCache();
     Transform& X_FM = toB(pc.bodyJointInParentJointFrame);
     X_FM.updR().setRotationToIdentityMatrix();
+
+    // Precalculate H_FM_F and H_PB_G here since they never change for a
+    // LoneParticle (the rotational part is zero and the translational part
+    // is identity). And since we require F=P=G here, these are the same.
+    Mat<2,3,Vec3>& H_PB_G = // a.k.a. just "H"
+        Mat<2,3,Vec3>::updAs(&pc.storageForH_PB_G[2 * uIndex]);
+    Mat<2, 3, Vec3>& H_FM_F =
+        Mat<2,3,Vec3>::updAs(&pc.storageForH_FM[2 * uIndex]);
+    H_PB_G.setToZero();
+    for (int j = 0; j < 3; ++j)
+      H_PB_G(1, j)[j] = 1;
+    H_FM_F = H_PB_G;
```

Comment:
`Mat<2,3,Vec3>::updAs(&pc.storageForH_FM[2 * uIndex]) = H_PB_G` if you want to drop a local that's unused for most of this logic